src/server/db_slice.cc

```diff
@@ -830,4 +843,24 @@ size_t DbSlice::EvictObjects(size_t memory_to_free, PrimeIterator it, DbTable* t
   return freed_memory_fun();
 };
 
+void DbSlice::RegisterWatchedKey(std::string_view key, ConnectionState::ExecInfo* exec_info) {
+  VLOG(0) << "Is now watched " << key;
+  auto it = watched_keys_.find(key);
+  if (it == watched_keys_.end()) {
+      it = watched_keys_.emplace(std::string(key)).first;
+  }
```

Comment:
We can't use a single emplace, because we'd have to construct a string from our sv

Reply:
It's true for other containers. See https://abseil.io/docs/cpp/guides/container  for the abseil flat hash map.

Google guys are smart.

Reply:
`Allows optimizations for emplace({key, value}) to avoid allocating a pair in most common cases.`

Reply:
also read the paragraph there "Heterogeneous Lookup"

Reply:
I see now. They don't list all the functions that support heterogeneous lookup in their docs (so they implicitly mean all of them). But do this in doc comments https://github.com/abseil/abseil-cpp/blob/0a066f31d981d69f7bde961055691906dabd4a3c/absl/container/flat_hash_map.h#L63-L65


STL does this for example only for find.